### ISSUE TITLE
paulgraham.com

### ISSUE THREAD

**Issue by Popolechien** [openzim/zim-requests#1143]
- Website URL: https://www.paulgraham.com
- License: permission - see https://kiwix.freshdesk.com/a/tickets/74506
- Desired ZIM Title: **Paul Graham's essays**
- Desired ZIM Description: **From the founder of Y Combinator**
- Desired ZIM Icon –png (URL or attach one): n/a
- Language (ISO 639-3): **eng**
- Is this a MediaWiki?: **no**

**Comment by RavanJAltaie** [comment]
Recipe created
https://farm.openzim.org/recipes/paulgraham.com_en_all
I'll update the library link once ready

**Comment by benoit74** [comment]
From @RavanJAltaie in wrong issue

> https://dev.library.kiwix.org/viewer#paulgraham.com_en_all_2024-08
The file is not complete, I've tried to change the scope type but same result.
Could you please check? @benoit74

**Comment by RavanJAltaie** [comment]
@benoit74 thanks for fixing the position of the comment @benoit74

**Comment by RavanJAltaie** [comment]
@benoit74 Yet any comment for the incomplete file?

**Comment by benoit74** [comment]
Sorry, looks like I forgot this issue after moving it.

Could you be more specific in what is missing? I don't see anything missing.

I however see two serious flaws:
- there is a constantly annoying popup saying "No store-id passed to the loader script" everytime we load a new page. Do you see it as well or am I the only one? This is going to be hard and fragile to fix AFAIK.
- the UI feels weird on desktops because it is optimized for small mobile screens and does not adapt dynamically ... I don't know if we really want to fix this ... if we switch to something optimized for desktops, then the look and feel will be weird / ugly on mobile screens. @kelson42 @Popolechien do you have any recommendation on this kind of situation? Should we create two ZIMs, one for desktop screens and one for mobile screens?

**Comment by Popolechien** [comment]
Well our policy for Wikimedia content is to aim for mobile (though I'm not sure what you mean by "small mobile"), so I'd focus on that one. But tbh I don't understand how it can display fine on a desktop browser and not so much on Kiwix-server.
![Screenshot 2024-09-17 at 09 16 37](https://github.com/user-attachments/assets/0945104b-65d4-41ae-b781-4c607f7caf9d)
![Screenshot 2024-09-17 at 09 17 10](https://github.com/user-attachments/assets/313b0c30-341e-495a-a5b4-569d4099ed2c)


I also get the annoying popup btw.

**Comment by benoit74** [comment]
> But tbh I don't understand how it can display fine on a desktop browser and not so much on Kiwix-server.

You are probably not at same page resolution and/or you did not reloaded  the page. And this does not look at all like a mobile screen resolution (you should aim for something like ~400px).

Let's showcase it with a video:

https://github.com/user-attachments/assets/b2b06810-a2cc-4722-bee9-23a3e67817a8

At the beginning of the video, I show the website on a desktop screen. This is perfectly readable.

Then I switch to a mobile screen. This is not readable at all. This is what would happen on mobile phones if we capture this desktop version.

Then I reload the page with still this mobile screen. The UI is completely updated to be readable on a mobile screen.

Then I switch to a desktop screen. This looks exactly like currently inside the ZIM when opened on a desktop screen.

**Comment by Popolechien** [comment]
Ok two things here:
- What kind of god-forsaken backend is that guy using that renders so poorly (no offense, but hey)? 
- What about the shopping cart being only viewable on desktop?

**Comment by benoit74** [comment]
`¯\_(ツ)_/¯`

See [FAQ](https://paulgraham.com/gfaq.html) - especially last question - to get a better grasp on who Paul Graham is ^^ 

I don't expect experts in Lisp and other esotheric languages to share my views on many things. Not saying any of us is right or wrong, or that we cannot work together, we just have different understanding of the world ^^

**Comment by tamnd** [comment]
Kage maintainer here.
https://github.com/tamnd/kage

I have already mirrored Paul Graham’s website to the ZIM format, and some Kiwix compatibility issues have now been fixed. Could we work more closely to bring this into openZIM?

**Comment by benoit74** [comment]
Would you mind to share how you achieved to make Paul Graham's website work in a ZIM, especially the issue of proper UI on both mobile and desktop? 

I assume the other problem was quite easy to get rid of by simply removing any JS script, which is kinda the opposite approach of zimit where we aim at ensuring all JS will work inside the ZIM.

**Comment by tamnd** [comment]
Here are the commands I use to clone and pack `paulgraham.zim`:

```bash
kage clone https://paulgraham.com/ --out ~/data/pg
```

and:

```bash
kage pack ~/data/pg/paulgraham.com --out ~/paulgraham.zim --title "Paul Graham" --description "Paul Graham essays, cloned by kage"
```

The result is a 1.6 MB ZIM file with 474 articles.

It works nicely in the Kiwix desktop app:

<img width="900" height="552" alt="Image" src="https://github.com/user-attachments/assets/b6981734-824d-4dd0-833b-d8e7a02532c7" />

<img width="900" height="553" alt="Image" src="https://github.com/user-attachments/assets/5749ad2b-dd57-487c-9edb-12aa9958a323" />

Let me grab my phone and try importing it into the Kiwix mobile app. : )

For detail technical how it works, I wrote here https://github.com/kiwix/overview/issues/162#issuecomment-4728925775

**Comment by tamnd** [comment]
<img width="1284" height="2585" alt="Image" src="https://github.com/user-attachments/assets/f9ef6279-c5c1-47a0-b1c2-5b446744f183" />
<img width="1252" height="2562" alt="Image" src="https://github.com/user-attachments/assets/01fef431-c236-4be3-8116-969c1694275c" />
<img width="1274" height="2570" alt="Image" src="https://github.com/user-attachments/assets/85265a44-b081-4df8-af3d-b886753f9aad" />

All links work nicely on mobile, but the text seems too small.

**Comment by benoit74** [comment]
Yeah, this is the problem I faced as well on mobile ... Text is definitely too small. Looks like CSS is dynamically chosen server-side based on User-Agent requesting the HTML. At least your approach allows to get rid of the nasty JS alert we have when using zimit.

**Comment by tamnd** [comment]
<img width="1284" height="2577" alt="Image" src="https://github.com/user-attachments/assets/2ccb7973-c376-4430-9450-57ec333fe8e2" />
<img width="1284" height="2572" alt="Image" src="https://github.com/user-attachments/assets/466778f2-3117-49a5-8002-34a9451bfed6" />

How about this? :)

I added the `--mobile` option to modify some CSS styles from the Paul Graham website, and for all old websites. The reasons are: no viewport meta tag, `<font size="2">` in the markup, etc.

The commands that I ran with Kage to reproduce (v0.3.6):

```bash
kage clone \
  --mobile \
  --all-asset-hosts \
  --force \
  --out $HOME/data/kage \
  https://paulgraham.com
```

```bash
kage pack \
  --out $HOME/data/kage/paulgraham.zim \
  $HOME/data/kage/paulgraham.com
```

The result is an 18MB ZIM file. With `--all-asset-hosts`, it fetched all images. Paul uses GIFs as title images.
